ports/stm32/adc.c

```diff
@@ -281,7 +281,7 @@ STATIC void adc_init_single(pyb_obj_adc_t *adc_obj) {
 
     adcx_init_periph(&adc_obj->handle, ADC_RESOLUTION_12B);
 
-#if defined(STM32L4)
+#if defined (STM32L471xx) || defined (STM32L475xx) || defined (STM32L476xx) || defined (STM32L485xx) || defined (STM32L486xx) || defined (STM32L496xx) || defined (STM32L4A6xx)
```

Comment:
Is there a better way to detect this feature instead of listing all supported MCUs?  Like a macro that is defined only if this feature is available (eg `ADC_MODE_INDEPENDENT`)?  Also, for consistency with rest of the code, please no space between `defined` and the open parenthesis.

Reply:
hmm, I agree it is better using such a switch than listing all MCUs. However, I copied this if-defined condition from ST's HAL library (stm32lib/STM32L4xx_HAL_Driver/Inc/stm32l4xx_hal_adc_ex.h) so I think ST currently does not provide any macro to explicitly show which ADC modes are available for now.